theseus/core/objective.py

```diff
@@ -195,11 +211,18 @@ def add(self, cost_function: CostFunction):
 
         self.cost_functions_for_weights[cost_function.weight].append(cost_function)
 
-        if self.optim_vars.keys() & self.aux_vars.keys():
-            raise ValueError(
-                "Objective does not support a variable being both "
-                "an optimization variable and an auxiliary variable."
-            )
+        for optim_name in self._new_optim_var_name:
+            if self.has_aux_var(optim_name):
+                raise ValueError(
+                    "Objective does not support a variable being both "
```

Comment:
Since you are reusing this message, better to define a variable called `dual_var_err_msg` and reference it in both `ValueError`.